merino-adm/src/remote_settings/mod.rs

```diff
@@ -296,6 +296,11 @@ impl RemoteSettingsSuggester {
 
                     let icon_key = format!("icon-{}", adm_suggestion.icon);
                     let icon_url = if let Some(u) = icon_urls.get(&icon_key) {
+                        // Suppress the warning as `u` does not live long enough
+                        // since `from_maybe_shared()` expects its argument to
+                        // be either a reference type with static lifetime or an
+                        // owned type.
+                        #[allow(clippy::unnecessary_to_owned)]
                         Uri::from_maybe_shared(u.to_string()).expect("invalid URL")
```

Comment:
I wonder if clippy's also confused about this one but I can't blame it, `from_maybe_shared` is a little confusing here. We can simplify it to `from_str(u)` w/ `use std::str::FromStr;` (or `try_from` which acts similarly but `from_str` reads better IMO) to avoid the clippy warning entirely.

```suggestion
                        Uri::from_str(u).expect("invalid URL")
```

Reply:
Ah, good point. Thanks!